engine/src/main/java/nl/inl/blacklab/searches/SearchCountFromResults.java

```diff
@@ -32,36 +26,18 @@
      */
     private final CountType type;
 
-    /**
-     * The (running or finished) result count.
-     * We can peek at this while it's running, or wait for executeInternal() to
-     * complete, returning the final results.
-     */
-    private ResultsStats resultCount;
-
-    /**
-     * Latch that marks the end of the search for this count.
-     */
-    private final CountDownLatch searchLatch;
-
-    /**
-     * Maximum time to wait for a search before giving up
-     */
-    private final int MAX_SEARCH_WAIT_IN_MS = 2000;
-
     public SearchCountFromResults(QueryInfo queryInfo, SearchForResults<T> source, CountType type) {
         super(queryInfo);
         this.source = source;
         this.type = type;
-        this.searchLatch = new CountDownLatch(1);
     }
 
     @Override
-    public ResultsStats executeInternal() throws InvalidQuery {
+    public ResultsStats executeInternal(Peekable<ResultsStats> progressReporter) throws InvalidQuery {
```

Comment:
Hey @jan-niestadt I wanted to run the following idea by you,  in case this could help us simplify this problem. What if in `executeInternal` we schedule a `processedTotal()` call to be run asynchronously and we return from this method as soon as the search is done.

We would also have to save the future performing the full count as part of this object

For dependents looking for the full count they can check on the future above to make sure the full account has been processed.  Otherwise we can return the resultStats object if not null  or 0 if it is null

Thoughts?

Reply:
What you're describing is essentially the way it used to work. Yes, `processedTotal()` could be executed asynchronously, but it'd have to be its own `Search` so it can be aborted if it's taking too long. So now there's two searches, with one depending on the other. This caused issues with the count seemingly being stuck forever because its corresponding `processedTotal()` search had been cancelled, but it had no way to detect that.

That said, it's probably worth investigating. I thought this version might be a simplication, but I agree with you that it's gotten quite complicated. If we can go back to the original design and fix the hanging problem, that could be better.

Getting rid of the `Delegate` class would be nice. We'd have to make sure code doesn't ever try to call `processedTotal()` on the running count. I'll give it a go today.

Reply:
I see, I probably miss the `waitfortotal=true` detail.  You know, I am with you, I'd say let's give this a try.

If we encounter problems we can revaluate. I can bring forward a set of changes that more or less implemented the idea the idea above.

Reply:
Great, I'll merge this now and we can hopefully simplify it later. Of course I'll be happy to take a look at your proposed version.